src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -71,11 +72,34 @@ private static IEndpointConventionBuilder CreateSubscribeEndPoint(IEndpointRoute
                     {
                         var topicMetadata = e.Metadata.GetOrderedMetadata<ITopicMetadata>();
                         var originalTopicMetadata = e.Metadata.GetOrderedMetadata<IOriginalTopicMetadata>();
+                        var bulkSubscribeMetadata = e.Metadata.GetOrderedMetadata<IBulkSubscribeMetadata>();
 
-                        var subs = new List<(string PubsubName, string Name, string DeadLetterTopic, bool? EnableRawPayload, string Match, int Priority, Dictionary<string, string[]> OriginalTopicMetadata, string MetadataSeparator, RoutePattern RoutePattern)>();
+                        var subs = new List<(string PubsubName, string Name, string DeadLetterTopic, bool? EnableRawPayload, 
+                            string Match, int Priority, Dictionary<string, string[]> OriginalTopicMetadata, 
+                            string MetadataSeparator, RoutePattern RoutePattern, DaprTopicBulkSubscribe bulkSubscribe)>();
 
                         for (int i = 0; i < topicMetadata.Count(); i++)
                         {
+                            DaprTopicBulkSubscribe bulkSubscribe = null;
+
+                            for (int j = 0; j < bulkSubscribeMetadata.Count(); j++)
```

Comment:
Think this could be a foreach instead of using `j`?

Reply:
Fixed